transports/tcp/src/lib.rs

```diff
@@ -475,12 +362,18 @@ where
         log::debug!("dialing {}", socket_addr);
 
         let socket = self
-            .create_socket(socket_addr)
+            .create_socket(socket_addr, opts.port_use)
             .map_err(TransportError::Other)?;
 
-        if let Some(addr) = self.port_reuse.local_dial_addr(&socket_addr.ip()) {
-            log::trace!("Binding dial socket to listen socket {}", addr);
-            socket.bind(&addr.into()).map_err(TransportError::Other)?;
+        match self.port_reuse.local_dial_addr(&socket_addr.ip()) {
+            Some(socket_addr) if opts.port_use == PortUse::Reuse => {
+                log::trace!("Binding dial socket to listen socket {}", socket_addr);
+                socket
+                    .bind(&socket_addr.into())
+                    .map_err(TransportError::Other)?;
+                self.port_reuse.dialed_from_listener(addr);
```

Comment:
I think this function name is too similar to our concept of "dialing as a listener" when it has nothing to do with it.

Reply:
Yes that's right, I hope you like the new name.